ruby/README.md

```diff
@@ -686,6 +686,39 @@ Examples:
 
 ### Try to make your classes as [SOLID](http://en.wikipedia.org/wiki/SOLID_(object-oriented_design\)) as possible.
 
+### Namespace Definition
+Define (and reopen) namespaced classes and modules using explicit nesting.
+Using the scope resolution operator can lead to surprising constant lookups due to Ruby's https://cirw.in/blog/constant-lookup.html[lexical scoping], which depends on the module nesting at the point of definition.
+
+```rb
+module Utilities
+  class Queue
+  end
+end
+
+# good
+module Utilities
+  class WaitingList
+    Module.nesting # => [Utilities::WaitingList, Utilities]
+
+    def initialize
+      @queue = Queue.new # Refers to Utilities::Queue
```

Comment:
Would it help to note the value of prefixing `::` to for escaping back to the top level? Or is it